IsoLib/libisomediafile/src/MP4Movies.c

```diff
@@ -75,6 +75,35 @@ MP4_EXTERN(MP4Err) MP4DisposeMovie(MP4Movie theMovie)
     dh->close(dh);
   }
 
+  if(moov->moovAtomPtr->type == MP4MovieFragmentAtomType)
+  {
+    MP4MovieFragmentAtomPtr moof;
+    MP4TrackFragmentAtomPtr traf;
+    MP4MovieAtomPtr movieAtom;
+    MP4TrackExtendsAtomPtr trex;
+    u32 i;
+    u32 trackTotalCount;
+
+    moof = (MP4MovieFragmentAtomPtr)moov->moovAtomPtr;
+    for(i = 0; i < moof->atomList->entryCount; i++)
+    {
+      err = MP4GetListEntry(moof->atomList, i, (char **)&traf);
+      if(err) goto bail;
+      traf->destroy((MP4AtomPtr)traf);
+    }
+
+    movieAtom       = (MP4MovieAtomPtr)moov->true_moov;
+    trackTotalCount = movieAtom->getTrackCount(movieAtom);
+    if(trackTotalCount != 0)
+    {
+      for(i = 1; i <= trackTotalCount; i++)
+      {
+        movieAtom->getTrackExtendsAtom(movieAtom, i, (MP4AtomPtr *)&trex);
+        trex->destroy((MP4AtomPtr)trex);
+      }
+    }
+  }
+
   if(moov->moovAtomPtr) moov->moovAtomPtr->destroy(moov->moovAtomPtr);
```

Comment:
Is the above not already handled by this call?

Reply:
If It makes movie atom for fragment mode, 
It need to free the memory traf atom and trex atom.

```
if(moov->moovAtomPtr) moov->moovAtomPtr->destroy(moov->moovAtomPtr);
```
This part do not free the memory of atoms derived from fragments